tests/baselines/reference/derivedClassSuperStatementPosition.errors.txt

```diff
@@ -0,0 +1,63 @@
+tests/cases/conformance/classes/constructorDeclarations/superCalls/derivedClassSuperStatementPosition.ts(12,15): error TS2785: A 'super' call must be a root-level statement within a constructor of a derived class that contains initialized properties, parameter properties, or private identifiers.
+tests/cases/conformance/classes/constructorDeclarations/superCalls/derivedClassSuperStatementPosition.ts(21,13): error TS2785: A 'super' call must be a root-level statement within a constructor of a derived class that contains initialized properties, parameter properties, or private identifiers.
+tests/cases/conformance/classes/constructorDeclarations/superCalls/derivedClassSuperStatementPosition.ts(33,13): error TS2785: A 'super' call must be a root-level statement within a constructor of a derived class that contains initialized properties, parameter properties, or private identifiers.
+tests/cases/conformance/classes/constructorDeclarations/superCalls/derivedClassSuperStatementPosition.ts(42,13): error TS2785: A 'super' call must be a root-level statement within a constructor of a derived class that contains initialized properties, parameter properties, or private identifiers.
+
+
+==== tests/cases/conformance/classes/constructorDeclarations/superCalls/derivedClassSuperStatementPosition.ts (4 errors) ====
+    class DerivedBasic extends Object {
+        prop = 1;
+        constructor() {
+            super();
+        }
+    }
+    
+    class DerivedInConditional extends Object {
+        prop = 1;
+        constructor() {
+            Math.random()
+                ? super(1)

```

Comment:
Perhaps for a future PR, but it would be nice to actually support this. Its technically possible:

```ts
// ts
class Derived extends Base {
  prop = 1;
  constructor() {
    Math.random() ? super(1) : super(0);
  }
}
// es2015
class Derived extends Base {
  constructor() {
    // wrap `super` and initialization in an arrow
    const _super = (...args) => {
      super(...args);
      this.prop = 1;
      return this;
    };

    // class body
    Math.random() ? _super(1) : _super(0);
  }
}
// es5
var Derived = /** @class */ (function (_super) {
  __extends(Derived, _super);
  function Derived() {
    var _this = this;
    // wrap `super` and initialization in a function expression
    var _super_1 = function() {
      _this = _super.apply(_this, arguments) || _this;
      _this.prop = 1;
      return _this;
    };

    // class body
    Math.random() ? _super_1(1) : _super_1(0);
  }
  return Derived;
}(Base));
```